diff --git a/drivers/misc/mediatek/connectivity/combo/common/core/include/wmt_conf.h b/drivers/misc/mediatek/connectivity/combo/common/core/include/wmt_conf.h
index 7c6f6750..de5aa815 100644
--- a/drivers/misc/mediatek/connectivity/combo/common/core/include/wmt_conf.h
+++ b/drivers/misc/mediatek/connectivity/combo/common/core/include/wmt_conf.h
@@ -19,7 +19,7 @@
 ********************************************************************************
 */
 #define CUST_CFG_WMT "WMT.cfg"
-#define CUST_CFG_WMT_PREFIX "/system/etc/firmware/"
+#define CUST_CFG_WMT_PREFIX "/lib/firmware/postmarketos/"
 
 /*******************************************************************************
 *                    E X T E R N A L   R E F E R E N C E S
diff --git a/drivers/misc/mediatek/connectivity/combo/common/linux/stp_chrdev_ant.c b/drivers/misc/mediatek/connectivity/combo/common/linux/stp_chrdev_ant.c
index e252ae69..2f12dad2 100644
--- a/drivers/misc/mediatek/connectivity/combo/common/linux/stp_chrdev_ant.c
+++ b/drivers/misc/mediatek/connectivity/combo/common/linux/stp_chrdev_ant.c
@@ -21,8 +21,8 @@ MODULE_LICENSE("Dual BSD/GPL");
 #define ANT_DRIVER_NAME "mtk_stp_ANT_chrdev"
 #define ANT_DEV_MAJOR 197	/* never used number */
 static PINT8 ANT_BUILT_IN_PATCH_FILE_NAME;
-static PINT8 ANT_BUILT_IN_PATCH_FILE_NAME_E1 = "/system/etc/firmware/ANT_RAM_CODE_E1.BIN";
-static PINT8 ANT_BUILT_IN_PATCH_FILE_NAME_E2 = "/system/etc/firmware/ANT_RAM_CODE_E2.BIN";
+static PINT8 ANT_BUILT_IN_PATCH_FILE_NAME_E1 = "/lib/firmware/postmarketos/ANT_RAM_CODE_E1.BIN";
+static PINT8 ANT_BUILT_IN_PATCH_FILE_NAME_E2 = "/lib/firmware/postmarketos/ANT_RAM_CODE_E2.BIN";
 
 #define PFX                         "[MTK-ANT] "
 #define ANT_LOG_DBG                  3
diff --git a/drivers/misc/mediatek/connectivity/combo/drv_wlan/mt6630/wlan/os/linux/gl_kal.c b/drivers/misc/mediatek/connectivity/combo/drv_wlan/mt6630/wlan/os/linux/gl_kal.c
index bce4edca..92a39639 100755
--- a/drivers/misc/mediatek/connectivity/combo/drv_wlan/mt6630/wlan/os/linux/gl_kal.c
+++ b/drivers/misc/mediatek/connectivity/combo/drv_wlan/mt6630/wlan/os/linux/gl_kal.c
@@ -883,6 +883,7 @@ static mm_segment_t orgfs;
 static PUINT_8 apucFwPath[] = { 
 	(PUINT_8) "/storage/sdcard0/",
 	(PUINT_8) "/etc/firmware/",
+	(PUINT_8) "/lib/firmware/postmarketos/",
 	NULL
 };
 
diff --git a/drivers/misc/mediatek/connectivity/conn_soc/common/core/include/wmt_conf.h b/drivers/misc/mediatek/connectivity/conn_soc/common/core/include/wmt_conf.h
index 7ad0fca5..0dd30b3f 100644
--- a/drivers/misc/mediatek/connectivity/conn_soc/common/core/include/wmt_conf.h
+++ b/drivers/misc/mediatek/connectivity/conn_soc/common/core/include/wmt_conf.h
@@ -33,7 +33,7 @@
 ********************************************************************************
 */
 #define CUST_CFG_WMT "WMT_SOC.cfg"
-#define CUST_CFG_WMT_PREFIX "/system/etc/firmware/"
+#define CUST_CFG_WMT_PREFIX "/lib/firmware/postmarketos/"
 
 /*******************************************************************************
 *                    E X T E R N A L   R E F E R E N C E S
diff --git a/drivers/misc/mediatek/eemcs/eemcs_expt.c b/drivers/misc/mediatek/eemcs/eemcs_expt.c
index aee65a62..a8e55472 100644
--- a/drivers/misc/mediatek/eemcs/eemcs_expt.c
+++ b/drivers/misc/mediatek/eemcs/eemcs_expt.c
@@ -1103,7 +1103,7 @@ void eemcs_expt_reset_statistics(void)
 }
 
 #define EEMCS_EXCEPTION_LOG_TO_FILE  1
-#define EEMCS_EXCEPTION_LOG_PATH    "/system/etc/firmware"
+#define EEMCS_EXCEPTION_LOG_PATH    "/lib/firmware/postmarketos"
 
 /*
  * @brief Flush all packets in exception instance to files for debugging
diff --git a/drivers/misc/mediatek/eemcs/eemcs_fs_ut.c b/drivers/misc/mediatek/eemcs/eemcs_fs_ut.c
index 827eb604..27ef2176 100644
--- a/drivers/misc/mediatek/eemcs/eemcs_fs_ut.c
+++ b/drivers/misc/mediatek/eemcs/eemcs_fs_ut.c
@@ -62,7 +62,7 @@
 #define MD_ROOT_DIR                     "Z:"
 #define DSP_ROOT_DIR                    "Y:"
 #define FS_ROOT_DIR                     "/data/nvram/md1"
-#define FS_DSP_ROOT_DIR                 "/system/etc/firmware"
+#define FS_DSP_ROOT_DIR                 "/lib/firmware/postmarketos"
 
 #define MAX_FS_PKT_BYTE                 (0x1000 - 128)
 #define CCCI_FS_REQ_SEND_AGAIN          0x80000000
diff --git a/drivers/misc/mediatek/mach/mt6795/include/mach/md32_helper.h b/drivers/misc/mediatek/mach/mt6795/include/mach/md32_helper.h
index f6a703b8..b5cedd51 100644
--- a/drivers/misc/mediatek/mach/mt6795/include/mach/md32_helper.h
+++ b/drivers/misc/mediatek/mach/mt6795/include/mach/md32_helper.h
@@ -42,8 +42,8 @@ extern struct md32_regs md32reg;
 #define MD32_JTAG_GPIO_DVT 0
 
 #ifdef CONFIG_MTK_LDVT
-  #define MD32_DATA_IMAGE_PATH "/system/etc/firmware/md32_d.bin"
-  #define MD32_PROGRAM_IMAGE_PATH "/system/etc/firmware/md32_p.bin"
+  #define MD32_DATA_IMAGE_PATH "/lib/firmware/postmarketos/md32_d.bin"
+  #define MD32_PROGRAM_IMAGE_PATH "/lib/firmware/postmarketos/md32_p.bin"
 #else
   #define MD32_DATA_IMAGE_PATH "/etc/firmware/md32_d.bin"
   #define MD32_PROGRAM_IMAGE_PATH "/etc/firmware/md32_p.bin"
diff --git a/drivers/misc/mediatek/wmt_ccci/wmt_cfg_parser.h b/drivers/misc/mediatek/wmt_ccci/wmt_cfg_parser.h
index 54aea826..4dea8714 100644
--- a/drivers/misc/mediatek/wmt_ccci/wmt_cfg_parser.h
+++ b/drivers/misc/mediatek/wmt_ccci/wmt_cfg_parser.h
@@ -56,7 +56,7 @@ extern unsigned int wmtCcciLogLvl;
 #endif
 
 #define WMT_CFG_FILE "WMT_SOC.cfg"
-#define WMT_CFG_FILE_PREFIX "/system/etc/firmware/"
+#define WMT_CFG_FILE_PREFIX "/lib/firmware/postmarketos/"
 
 /*******************************************************************************
 *                             D A T A   T Y P E S
